From 3e7bd2e3b9607c5fcb786a0a083b106695c35275 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 10 Sep 2017 16:31:50 +0200
Subject: [PATCH 1/1] elf2efi.c: ignore R_ARM_V4BX

Relocation type R_ARM_V4BX requires no computation.
It marks the location of an ARMv4 branch exchange instruction.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 tools/elf2efi.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tools/elf2efi.c b/tools/elf2efi.c
index 97a2126b92..e2a78f33d5 100644
--- a/tools/elf2efi.c
+++ b/tools/elf2efi.c
@@ -651,6 +651,9 @@ static void process_reloc(struct elf_file *elf, const Elf_Shdr *shdr,
 			 * loaded.
 			 */
 			break;
+		case ELF_MREL(EM_ARM, R_ARM_V4BX):
+			/* Skip instructions without computation */
+			break;
 		default:
 			eprintf("Unrecognised relocation type %d\n", type);
 			exit(1);
-- 
2.11.0

